junit-jupiter-api/src/main/java/org/junit/jupiter/api/Enabled.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import static org.apiguardian.api.API.Status.STABLE;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.apiguardian.api.API;
+
+/**
+ * {@code @Enabled} is used to signal that the annotated test class or
+ * test method is only executed if and only if the script passed via
+ * the {@link #value()} property evaluates to {@code true}.
+ *
+ * <p>When applied at the class level, all test methods within that class
+ * are automatically disabled as well.
+ *
+ * @since 5.1
+ */
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@API(status = STABLE, since = "5.1")
+public @interface Enabled {
+
+	/**
+	 * Short name of the {@link javax.script.ScriptEngine ScriptEngine} to use.
+	 *
+	 * @return script engine name, defaults to {@code "javascript"}
+	 * @see javax.script.ScriptEngineManager#getEngineByName(String)
+	 */
+	String engine() default "javascript";
```

Comment:
I think we should make the default engine a configuration parameter because it's very likely that you'd want to use the same engine for all your `@Enabled` annotations.

Reply:
Good point. If available, I'll prefer the configuration parameter value over the default return value defined here.